internal/provider/kubernetes/controller.go

```diff
@@ -1613,6 +1669,17 @@ func (r *gatewayAPIReconciler) processGatewayClassParamsRef(ctx context.Context,
 	if err := r.processEnvoyProxy(ep, resourceMap); err != nil {
 		return err
 	}
+
+	if classRefsEnvoyProxy(gc, ep) {
+		if err := r.addFinalizer(ctx, ep); err != nil {
+			return fmt.Errorf("failed to add finalizer to Envoy Proxy %s: %w", ep.Name, err)
+		}
+		if err := r.addFinalizer(ctx, gc); err != nil {
```

Comment:
we should not add finalizer for gc at this moment, nor it will add `gateway-exists-finalizer.gateway.networking.k8s.io` finalizer for those gc that have no gateways attach to it.

we should this finalizer for those gc that have gateways attach to it.